src/main/scala/cromwell/engine/backend/runtimeattributes/CromwellRuntimeAttributes.scala

```diff
@@ -24,163 +29,187 @@ case class CromwellRuntimeAttributes(docker: Option[String],
+    defaultAttributes ++ taskAttributes
   }
 
-  private def fromAttributeMap(attributeMap: AttributeMapTrait, backendType: BackendType): CromwellRuntimeAttributes = {
+  def validateKeys(wdlRuntimeAttributes: RuntimeAttributes, backendType: BackendType): Try[RuntimeAttributes] = {
     /**
       *  Warn if keys are found that are unsupported on this backend.  This is not necessarily an error, at this point
       *  the keys are known to be supported on at least one backend other than this.
       */
-    attributeMap.unsupportedKeys(backendType).foreach(unsupportedKeyMessage => log.info(unsupportedKeyMessage))
-
-    val attributeMapNel = validateRequiredKeys(attributeMap, backendType)
-    val runtimeAttributeNel = validateRuntimeAttributes(attributeMap)
-    val validatedRuntimeAttributes = (attributeMapNel |@| runtimeAttributeNel) { (_, r) => r }
-
-    validatedRuntimeAttributes match {
-      case Success(r) => r
-      case Failure(f) =>
-        throw new IllegalArgumentException() with ThrowableWithErrors {
-          val message = "RuntimeAttribute is not valid."
-          val errors = f
-        }
+    unsupportedKeys(wdlRuntimeAttributes.attrs.keys, backendType) foreach log.warn
+    validateRequiredKeys(wdlRuntimeAttributes.attrs, backendType) match {
+      case scalaz.Success(x) => scala.util.Success(wdlRuntimeAttributes)
+      case scalaz.Failure(e) => scala.util.Failure(new IllegalArgumentException() with ThrowableWithErrors {
+        val message = "RuntimeAttributes are invalid."
+        val errors = e
+      })
     }
   }
 
+  def unsupportedKeys(keys: Iterable[String], backendType: BackendType): Option[String] = {
+    val supportedKeys = backendType.supportedKeys map { _.key }
+    val unsupportedKeys = keys.toSet -- supportedKeys
+
+    if (unsupportedKeys.isEmpty) None
+    else Option(s"Found unsupported keys for backend '$backendType': " + unsupportedKeys.toSeq.sorted.mkString(", "))
+  }
+
   val LocalDiskName = "local-disk"
-  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, Option(10)).toDisk
-
-  /** Fallback values if values for these keys are not specified in a task "runtime" stanza. */
-  object Defaults {
-    val Cpu = 1L
-    val Disk = Vector(LocalizationDisk)
-    val FailOnStderr = false
-    val ContinueOnReturnCode = ContinueOnReturnCodeSet(Set(0))
-    val Memory = 2.0
-    val Preemptible = 0
-    val Zones = Vector("us-central1-a")
-  }
-
-  private def validateRuntimeAttributes(attributeMap: AttributeMapTrait): ValidationNel[String, CromwellRuntimeAttributes] = {
-    val docker = attributeMap.get(DOCKER)
-    val zones = attributeMap.get(ZONES) map { _.split("\\s+").toVector } getOrElse Defaults.Zones
+  val LocalizationDisk = LocalDisk(LocalDiskName, DiskType.SSD, sizeGb=10)
+
+  private val keys = Set("cpu", "disks", "docker", "zones", "continueOnReturnCode", "failOnStderr", "preemptible", "memory")
+
+  private val defaultValues = Map(
```

Comment:
> Shouldn't it be configurable?

I'm guessing you mean "configurable via the application.conf file"?  I agree, but given the impending conflicts we'll face, perhaps we should wait?